app/controllers/concerns/authenticatable.rb

```diff
@@ -0,0 +1,51 @@
+module Authenticatable
+  SSO_JWT_SESSION_KEY = :sso_session
+  SSO_JWT_REFRESH_KEY = :sso_refresh
+  SSO_JWT_USER_ID_KEY = :sso_user_id
+
+  def skip_authentication
+    if login_by_sso?
+      authenticate_sso_request!
+      redirect_to after_sign_in_path_for(:user) if current_user
+      return
+    end
+
+    if email_authentication_signed_in?
+      redirect_to after_sign_in_path_for(:user)
+    end
+  end
+
+  def authenticate_sso_request!
+    return unless login_by_sso?
+
+    st, rt = session[SSO_JWT_SESSION_KEY], session[SSO_JWT_REFRESH_KEY]
+    result = Accounts::SsoAuthentication.validate_or_refresh_session(st, rt)
+
+    # TODO: path where user has externally logged out, log them out of here as well
```

Comment:
Even though this TODO exists, I think this is already handled because of [this](https://github.com/tramlinehq/tramline/blob/df6a8a953990fb606bb03db182500eb3b6064b4a/app/controllers/signed_in_application_controller.rb#L56). Unconfirmed.